webapp/src/Controller/API/ProblemController.php

```diff
@@ -381,7 +418,7 @@ protected function getQueryBuilder(Request $request): QueryBuilder
             ->from(ContestProblem::class, 'cp')
             ->join('cp.problem', 'p')
             ->leftJoin('p.testcases', 'tc')
-            ->select('cp, partial p.{probid,externalid,name,timelimit,memlimit}, COUNT(tc.testcaseid) AS testdatacount')
+            ->select('cp, partial p.{probid,externalid,name,timelimit,memlimit,problemtext_type,problemtext}, COUNT(tc.testcaseid) AS testdatacount')
```

Comment:
So does this mean that now any problem related API call will download the full binary pdf from the database or for all problems for a list action) when this is only needed in the case that one requests one single statement explicitly?

Reply:
Hmmm yes. let me change that to only loading the type

Reply:
Note we do lease the feature that we expose the `hash`, but that's fine (and if we want this, we should store the hash as a column)